app/facades/websocket/WebSocketController.scala

```diff
@@ -150,20 +167,33 @@ class WebSocketController @Inject() (actorSystem:  ActorSystem,
   /**
    * Establishes a WebSocket connection.
    */
-  def dslinkWSConnect = WebSocket.acceptOrResult[DSAMessage, DSAMessage] { request =>
+  def dslinkWSConnect = acceptOrResult { sessionInfo =>
     import Settings.WebSocket._
 
-    log.debug(s"WS request received: $request")
-    val dsId = getDsId(request)
-    val sessionInfo = cache.get[DSLinkSessionInfo](dsId)
-    log.debug(s"Session info retrieved for $dsId: $sessionInfo")
-
     sessionInfo map { si =>
       createWSFlow(si, actors.downstream, BufferSize, OnOverflow) map Right[Result, DSAFlow]
     } getOrElse
       Future.successful(Left[Result, DSAFlow](Forbidden))
+  }
 
-  }(transformer)
+  private def acceptOrResult(f: Option[DSLinkSessionInfo] => Future[Either[Result, Flow[DSAMessage, DSAMessage, _]]]): WebSocket = {
+    WebSocket { request =>
+      log.debug(s"WS request received: $request")
```

Comment:
Minor, but @uralian avoids using string interpolation in debug to keep some performance. Maybe would be better to use standart java behavior:
`log.debug("WS request received: {}", request`

Reply:
yep, he does...

Reply:
As play.api.Logger is used here, we are not able to use the 'log.debug("WS request received: {}", request)' style, or maybe i don't know how. So let's leave as it is here.
There is interesting article about string literal performance in scala: https://medium.com/@dkomanov/scala-string-interpolation-performance-21dc85e83afd. There is a custom made macros there wich makes good performance improvement.

Reply:
it's ok, not a big deal